src/components/ContactsAdd.js

```diff
@@ -10,22 +10,99 @@ function ContactsAdd(props) {
 
   //TODO: Implement controlled form
   //send POST to json server on form submit
+  const formReset = {
+    firstName: '',
+    lastName: '',
+    street: '',
+    city: '',
+    email: '',
+    linkedIn: '',
+    twitter: ''
+  }
+  const [form, setForm] = useState(formReset)
+
+  const inputs = (e) => {
+    const formCopy = { ...form }
+    //const formValues = ['firstName','lastName','street','city']
+
+    if (e.target.id === 'firstName') {
```

Comment:
I think:

formCopy[e.target.id] = e.target.value

would work as a refactoring here and you wouldn't need all the if statements.